src/fault/fiber/thread.cpp

```diff
@@ -0,0 +1,90 @@
+#include <yaclib/fault/detail/fiber/thread.hpp>
+#include <yaclib/log.hpp>
+
+namespace yaclib::detail::fiber {
+
+using namespace std::chrono_literals;
+
+Thread::Thread() noexcept = default;
+
+void Thread::swap(Thread& t) noexcept {
+  auto* other_impl = t._impl;
+  auto other_queue = std::move(t._join_queue);
+  t._join_queue = std::move(_join_queue);
+  _join_queue = std::move(other_queue);
+  t._impl = _impl;
+  _impl = other_impl;
+  _joined_or_detached = std::exchange(t._joined_or_detached, _joined_or_detached);
+}
+
+bool Thread::joinable() const noexcept {
+  return !_joined_or_detached &&
+         (Scheduler::Current() == nullptr || Scheduler::Current()->GetId() != this->_impl->GetId());
+}
+
+void Thread::join() {
+  YACLIB_ERROR(_joined_or_detached, "already joined or detached on join");
+  _joined_or_detached = true;
+  // TODO(myannyax): allow joining from other threads
+  while (_impl->GetState() != Completed) {
+    _join_queue.Wait();
+  }
+  AfterJoinOrDetach();
+}
+
+void Thread::detach() {
+  _joined_or_detached = true;
+  AfterJoinOrDetach();
+}
+
+Fiber::Id Thread::get_id() const noexcept {
+  return _impl == nullptr ? Fiber::Id{0} : _impl->GetId();
+}
+
+Thread::native_handle_type Thread::native_handle() noexcept {
+  YACLIB_ERROR(true, "native_hande is not supported for fibers");
+}
+
+unsigned int Thread::hardware_concurrency() noexcept {
+  return std::thread::hardware_concurrency();
```

Comment:
Hmmm, maybe should be setted by user (default is ok)?

Reply:
I created method that can set it for fibers, to do it for threads I need to create thread wrapper and maybe it should be done in another pr?

Reply:
I don't think we need to make it for os threads